crates/ra_hir_def/src/adt.rs

```diff
@@ -57,18 +60,51 @@ pub struct FieldData {
     pub visibility: RawVisibility,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ReprKind {
+    Packed,
+    Other(TokenTree),
+}
+
+fn repr_from_value<T: AttrsOwner>(ast: &T) -> Option<ReprKind> {
+    ast.attrs().find_map(|a| {
+        a.as_simple_call().and_then(|(s, tt)| {
+            if s == "repr" {
+                let (tt, _) = ast_to_token_tree(&tt)?;
+                parse_repr_tt(&tt)
+            } else {
+                None
+            }
+        })
+    })
+}
+
+fn parse_repr_tt(tt: &Subtree) -> Option<ReprKind> {
+    let mut it = tt.token_trees.iter();
+    match it.next() {
+        None => None,
+        Some(TokenTree::Leaf(Leaf::Ident(ident))) if ident.text == "packed" => {
+            Some(ReprKind::Packed)
+        }
+        Some(tt) => Some(ReprKind::Other(tt.clone())),
+    }
+}
+
```

Comment:
@matklad I wanted to review this in draft first since I feel this is the wrong location for this parsing code, but I wasn't sure where to put it. `ra_cfg` seems to be where the `cfg` parsing code lives, but I wasn't sure if it would be overkill to introduce `ra_repr` for this. Also, should I go ahead and add support for all repr kinds at the moment, or do that as a follow-up?

Reply:
To me it seems fine to keep this code here for now. And I wouldn't add support for all repr kinds until we actually need it.